src/components/Grid.jsx

```diff
@@ -3,51 +3,38 @@
  */
 
 import React, { PropTypes } from 'react'
+import R from 'ramda'
 
 const propTypes = {
   horizontal: PropTypes.bool,
-  hourLength: PropTypes.number.isRequired,
-  timelineOffset: PropTypes.number.isRequired,
+  hours: PropTypes.number.isRequired,
+  offset: PropTypes.number.isRequired,
   color: PropTypes.string.isRequired,
 }
 
-const PATTERN_ID = 'Grid-pattern'
-
 class Grid extends React.Component {
 
-  getPattern () {
-    const hourOffset = this.props.timelineOffset * this.props.hourLength
-
-    if (this.props.horizontal) {
-      return (
-        <pattern id={PATTERN_ID} x={hourOffset} y="0" width={this.props.hourLength} height="100%">
-          <line x1="0" y1="0" x2="0" y2="100%" stroke={this.props.color} strokeWidth="1" />
-        </pattern>
-      )
-    } else {
-      return (
-        <pattern id={PATTERN_ID} x="0" y={hourOffset} width="100%" height={this.props.hourLength}>
-          <line x1="0" y1="0" x2="100%" y2="0" stroke={this.props.color} strokeWidth="1" />
-        </pattern>
-      )
-    }
+  getHorizontalLine (n) {
+    const pos = n / this.props.hours * 100 + (this.props.offset * 100 / this.props.hours)
+    return <line key={`gridline${n}`} x1={`${pos}%`} y1="0" x2={`${pos}%`} y2="100%"
+                 stroke={this.props.color} strokeWidth="1"/>
+  }
+
+  getVerticalLine (n) {
+    const pos = n / this.props.hours * 100 + (this.props.offset * 100 / this.props.hours)
+    return <line key={`gridline${n}`} x1="0" y1={`${pos}%`} x2="100%" y2={`${pos}%`}
+                 stroke={this.props.color} strokeWidth="1"/>
```

Comment:
No problem, just that @jnv has showed it on this exact piece of code, maybe it’s still sketched on your whiteboard…